src/wasm/wasm.cpp

```diff
@@ -973,12 +973,10 @@ void MemoryGrow::finalize() {
 void RefNull::finalize(HeapType heapType) { type = Type(heapType, true); }
 
 void RefNull::finalize(Type type_) {
-  assert(type_ == Type::unreachable || type_.isNullable());
```

Comment:
We shouldn't be asserting on stuff like this, we should be checking it in the validator. (We don't atm because nullability is a wip, though - removing this line matches the behavior there.)